Sources/hostmgr-helper/VirtualMachineSlot.swift

```diff
@@ -19,121 +19,160 @@ class VirtualMachineSlot: NSObject, ObservableObject {
+        try await virtualMachine.start()
+
+        let ipAddress: IPv4Address
+        if launchConfiguration.waitForNetworking {
+            ipAddress = try await vmManager.ipAddress(forVmWithName: launchConfiguration.handle)
+            Logger.helper.log("Startup complete – IP Address: \(ipAddress.debugDescription).")
+        } else {
+            Logger.helper.log("Startup in progress – skipped waiting for IP address per launch configuration.")
+            ipAddress = .any
         }
-        if let machine = managedVirtualMachine?.machine, machine.canStop {
-            do {
-                try await machine.stop()
-            } catch {
-                Logger.helper.error("Failed to stop VM: \(error)")
+        return ManagedVirtualMachine(machine: virtualMachine, config: launchConfiguration, ip: ipAddress)
+    }
+
+    private func stopManagedVm(_ mvm: ManagedVirtualMachine) async {
+        Logger.helper.log("Stopping VM \(mvm.handle).")
+        /// Quit responding to delegate methods
+        mvm.machine.delegate = nil
+        do {
+            if mvm.machine.canStop {
+                try await mvm.machine.stop()
+                Logger.helper.log("Stopped VM \(mvm.handle).")
             }
+        } catch {
+            Logger.helper.error("Failure when stopping VM: \(error)")
         }
-        await resetSlot()
     }
 
-    /// Resets the slot to a stopped status
-    ///
-    /// No need to do anything except some internal bookkeeping
-    /// - Parameter error: Error supplied if the VM crashed
-    @MainActor
-    func resetSlot(withError error: Error? = nil) async {
-        if let managedVirtualMachine {
-            do {
-                try await vmManager.removeVM(name: managedVirtualMachine.config.handle)
-            } catch {
-                Logger.helper.error("Failed to remove VM file: \(error)")
-            }
-            self.managedVirtualMachine = nil
+    private func cleanManagedVm(_ handle: String) async {
+        Logger.helper.log("Cleaning up VM \(handle).")
+        do {
+            try await vmManager.removeVM(name: handle)
+            Logger.helper.log("Cleaned up VM \(handle).")
+        } catch {
+            Logger.helper.error("Failure when removing VM files: \(error)")
+        }
+    }
+
+    func stop(withError error: Error? = nil) async throws {
+        Logger.helper.log("Stopping \(role.displayName) slot with current status \(status).")
+        switch status {
+        case .starting(let config, let task):
+            self.status = .stopping(config)
+            task.cancel()
+            await cleanManagedVm(config.handle)
+        case .running(let mvm):
+            self.status = .stopping(mvm.config)
+            await stopManagedVm(mvm)
+            await cleanManagedVm(mvm.handle)
+        default:
+            /// For all other states we do nothing.
+            Logger.helper.debug("Stop called while state was \(status) - doing nothing.")
```

Comment:
```suggestion
            // For all other states we do nothing.
            Logger.helper.debug("Stop called while state was \(status) - doing nothing.")
```

`///` generates documentation for types, methods, etc.

The way I see the comment here, it's an explanatory comment rather than a documentation one.